src/dragonfly/dragonfly/logger.py

```diff
@@ -53,15 +52,14 @@ def updateLED(self):
     def callback(self, data):
         self.position = data
         self.updateStatus(position=True)
-        self.time_offset = datetime.fromtimestamp(data.header.stamp.sec) - datetime.now()
```

Comment:
The benefit of using an offset instead of the time from the cube directly is that we always have a time st object to log again - even if it's wrong.    Thus, we wouldn't need conditional cases when we don't have a time value yet.  Also, I'm not sure how often time_reference is broadcast but using an offset means sub-broadcast-interval time will have the appropriate resolution for the given log.  Unless I'm missing something I'd like to see us use the time offset approach still.

Reply:
okay I'll put it back

Reply:
Okay pushed so now logs when powering up look like this here is the transitions between when it has no offset to when it gets the first time_reference msg
```
2022-09-21 01:01:47.282141 co2: 499.6600036621094 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ -
2022-09-21 01:01:47.378795 co2: 499.6700134277344 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ -
2022-09-21 01:01:47.484967 co2: 499.6700134277344 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ -
2022-09-21 01:01:47.572691 co2: 499.6700134277344 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ -
2022-09-21 01:01:47.683779 co2: 499.6700134277344 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ -
2022-09-21 01:01:47.782458 co2: 499.6700134277344 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ -
2022-09-21 01:01:47.876598 co2: 499.6700134277344 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ -
2022-09-21 01:01:47.989276 co2: 499.6700134277344 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ -
2022-09-21 01:01:48.084605 co2: 499.6700134277344 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ -
2022-09-21 01:01:48.180164 co2: 499.6700134277344 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ -
2022-09-21 01:01:48.276772 co2: 499.67999267578125 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ 35.0820623 -106.6252515 1557.0789379114335
2022-11-15 20:27:37.059636 co2: 499.67999267578125 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ 35.0820623 -106.6252513 1557.0789383363829
2022-11-15 20:27:37.152241 co2: 499.67999267578125 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ 35.0820623 -106.6252512 1557.0889385488579
2022-11-15 20:27:38.071977 co2: 499.69000244140625 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ 35.0820624 -106.6252511 1557.0889389108042
2022-11-15 20:27:38.169401 co2: 499.69000244140625 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ 35.0820624 -106.6252509 1557.0889393357543
2022-11-15 20:27:38.075483 co2: 499.69000244140625 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ 35.0820624 -106.6252507 1557.0889397607045
2022-11-15 20:27:38.172495 co2: 499.70001220703125 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ 35.0820624 -106.6252502 1557.0889408230817
2022-11-15 20:27:38.071900 co2: 499.70001220703125 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ 35.0820622 -106.6252498 1557.0889413740426
2022-11-15 20:27:38.162796 co2: 499.70001220703125 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ 35.082062 -106.6252495 1557.0789417125288
2022-11-15 20:27:38.045124 co2: 499.7099914550781 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ 35.0820618 -106.6252491 1557.0789422634928
2022-11-15 20:27:38.127836 co2: 499.7099914550781 55.0 7.699999809265137 34.5 851 55.0 55.0 0 @ 35.0820615 -106.6252488 1557.0889424525112

```